### PR TITLE
Add Docker API E2E tests for scaling nodes and worker node groups

### PR DESCRIPTION
**Pull request by cxbrowne1207**
*Issue #, if available:*

*Description of changes:*
This PR adds an e2e test for the docker API  or scaling nodes and add/remove worker node group configurations.

*Testing (if applicable):*
`--- PASS: TestDockerUpgradeKubernetes123to124WorkloadClusterScaleupAPI (2017.01s)
`
`--- PASS: TestDockerUpgradeWorkloadClusterScaleAddRemoveWorkerNodeGroupsAPI (1167.16s)
`
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/e2e/workload_api_test.go [hunk 1]
```diff
@@ -204,8 +204,7 @@ func TestDockerUpgradeKubernetes123to124WorkloadClusterScaleWorkerNodesAPI(t *te
 				api.WithManagementCluster(managementCluster.ClusterName),
 				api.WithControlPlaneCount(1),
 				api.RemoveAllWorkerNodeGroups(), // This gives us a blank slate
-				api.WithWorkerNodeGroup("worker-0", api.WithCount(2)),
-				api.WithWorkerNodeGroup("worker-1", api.WithCount(1)),
+				api.WithWorkerNodeGroup("worker-0", api.WithCount(1)),
```
Comment (taneyland) [inline test/e2e/workload_api_test.go:207]: Why are we changing this to remove the scale down case?

Comment (cxbrowne1207) [reply test/e2e/workload_api_test.go:207]: This one was changed to primarily test the k8s upgrade, so I figured scaling up is all that's needed. The new test added tests scaling up and down, as well as adding / remove worker node groups,

File: test/e2e/workload_api_test.go [hunk 2]
```diff
@@ -262,3 +260,44 @@ func TestDockerUpgradeWorkloadClusterLabelsAndTaintsAPI(t *testing.T) {
...
+	test := framework.NewMulticlusterE2ETest(t, managementCluster)
+	test.WithWorkloadClusters(
+		framework.NewClusterE2ETest(
+			t, provider, framework.WithClusterName(test.NewWorkloadClusterName()),
+		).WithClusterConfig(
+			api.ClusterToConfigFiller(
+				api.WithKubernetesVersion(v1alpha1.Kube124),
+				api.WithManagementCluster(managementCluster.ClusterName),
+				api.WithControlPlaneCount(1),
+				api.RemoveAllWorkerNodeGroups(), // This gives us a blank slate
```
Comment (mitalipaygude) [inline test/e2e/workload_api_test.go:285]: Out of curiosity, why do we need to remove all worker nodes?

Comment (cxbrowne1207) [reply test/e2e/workload_api_test.go:285]: There is a worker node group defined by default in the base configurations. So removing all worker node groups gives a clean slate.

**Pull request merged**